Update tutorial for latest versions of CUDA (#4229)
PointCloudLibrary/pcl: doc/tutorials/content/gpu_install.rst

Configuring your PC to use your Nvidia GPU with PCL
---------------------------------------------------
In this tutorial you will learn how to make your configure your system to make it compatible to run the GPU methods provided by PCL.
This tutorial is for Ubuntu, other Linux distrubutions can follow a similar process to set it up.  

> Windows is nto currently officialy supported for the GPU methods.

Reviewer: `not`

> The Explanation

Reviewer: Maybe rename to `Installing dependencies` and add a section for CUDA and another for Eigen

Author: I added a few sections. Is that fine?

Reviewer: Could you also mention that code works with clang (9 (or whatever is on 18.04), min version unknown) and GCC (min version 7) instead of saying works only with GCC 7 or later

Author: Added

---------------

In order to run the code you will need a system with an Nvidia GPU, having CUDA Toolkit v9.2+ installed. 
We will not be covering CUDA toolkit installation in this tutorial as there alredy exists many great tutorials for the same.
You can check your CUDA toolkit version using the following command:
 $ nvcc --version | grep "release" | awk '{print $6}' | cut -c2-

> The GPU methods in PCL only work with GCC 7 onwards. This will not be a problem if you are running Ubuntu 18+, in case you are running Ubuntu 16, you will need to install GCC 7 as the default one available is GCC 5.

Reviewer: Any limit on clang?

Author: Not aware as far as I remember. If I come across it will update the guide.

You can check your GCC version using the following command:

 $ gcc -dumpversion
 
To install GCC 7 run the following commands
 

>  $ sudo add-apt-repository ppa:ubuntu-toolchain-r/tesT

Reviewer: `test`

Configuring your PC to use your Nvidia GPU with PCL
---------------------------------------------------

> In this tutorial you will learn how to make your configure your system to make it compatible to run the GPU methods provided by PCL.

Reviewer 2: ```suggestion
In this tutorial you will learn how to configure your system to make it compatible to run the GPU methods provided by PCL.
```

This tutorial is for Ubuntu, other Linux distrubutions can follow a similar process to set it up.  

> Windows is **not** currently officialy supported for the GPU methods.

Reviewer 2: ```suggestion
Windows is **not** currently officially supported for the GPU methods.
```

Checking CUDA Version
---------------

In order to run the code you will need a system with an Nvidia GPU, having CUDA Toolkit v9.2+ installed. 

> We will not be covering CUDA toolkit installation in this tutorial as there alredy exists many great tutorials for the same.

Reviewer 2: ```suggestion
We will not be covering CUDA toolkit installation in this tutorial as there already exists many great tutorials for the same.
```

In order to run the code you will need a system with an Nvidia GPU, having CUDA Toolkit v9.2+ installed. 
We will not be covering CUDA toolkit installation in this tutorial as there alredy exists many great tutorials for the same.

You can check your CUDA toolkit version using the following command::

 $ nvcc --version | grep "release" | awk '{print $6}' | cut -c2-
 
 

> Checking GCC or Clang Version

Reviewer: Checking C++ Compiler version

---------------

The GPU methods in PCL require a min version of GCC 7 and work on Clang 6 onwards (min version unknown). 

> This will not be a problem if you are running Ubuntu 18+, in case you are running Ubuntu 16, you will need to install GCC 7 as the default one available is GCC 5.

Reviewer: This line needs rewording to incorporate clang

Author: Added some information on Clang

 $ nvcc --version | grep "release" | awk '{print $6}' | cut -c2-
 
 
Checking C++ Version
---------------

The GPU methods in PCL require a min version of GCC 7 or Clang 6 onwards (min version unknown). 

> This will not be a problem if you are running Ubuntu 18+, in case you are running Ubuntu 16, you will need to install GCC 7 or Clang 6 (lower versions not tested) as the default one available is GCC 5 and Clang 3.8.

Reviewer: ```suggestion
This will not be a problem if you are running Ubuntu 18.04 or later. However on Ubuntu 16.04, you will need to install GCC 7 or Clang 6 (lower versions not tested) manually because the versions available by default are: GCC 5 and Clang 3.8
```
